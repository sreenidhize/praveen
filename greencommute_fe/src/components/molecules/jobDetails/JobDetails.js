import React from "react";
import { Divider, Grid, Typography, makeStyles } from "@material-ui/core";
import PropTypes from "prop-types";

function JobDetails(props) {
  const useStyle = makeStyles((theme) => ({
    headingClass: {
      ...theme.navTabs,
    },
    ulClass: {
      listStylePosition: "initial",
      paddingLeft: "4%",
    },
    liClass: {
      paddingBottom: "3%",
    },
    descClass: {
      width: "20.5rem",
    },
    hrTop: {
      marginTop: "3%",
    },
    hrBottom: {
      marginBottom: "3%",
    },
  }));

  const { description, skills, ...rest } = props;

  const classes = useStyle();

  return (
    <Grid container spacing={4} style={{ width: "390px" }}>
      <Grid item xs={12}>
        <Divider variant="fullWidth" className={classes.hrBottom} />
      </Grid>
      <Grid item xs={12}>
        {/* <hr /> */}
        <Typography className={classes.headingClass} {...rest}>
          Description
        </Typography>
      </Grid>
      <Grid item xs={12}>
        <Typography variant="h6" className={classes.descClass}>
          {description}
        </Typography>
      </Grid>
      <Grid item xs={12}>
        <Divider variant="fullWidth" className={classes.hrTop} />
      </Grid>
      <Grid item xs={12}>
        <Typography className={classes.headingClass}>What it takes</Typography>
        <ul className={classes.ulClass}>
          {skills &&
            skills.map((skill, index) => (
              <li key={index} className={classes.liClass}>
                <Typography variant="h6" className={classes.descClass}>
                  {skill}
                </Typography>
              </li>
            ))}
        </ul>
      </Grid>
    </Grid>
  );
}

JobDetails.propTypes = {
  description: PropTypes.string,
  skills: PropTypes.array,
};

export default JobDetails;
