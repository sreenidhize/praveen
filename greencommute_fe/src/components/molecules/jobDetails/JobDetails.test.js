import React from "react";
import JobDetails from "./JobDetails";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("Job Details renders correctly", () => {
  const JObDetailsComponent = (props) => {
    return (
      <JobDetails
        data-testid={props.dataTestid}
        description="Open Text is seeking a talented..."
        skills={[
          "Excellent written and oral communication and presentation skills",
        ]}
      />
    );
  };
  test("should match snapshot", () => {
    const { asFragment } = render(<JObDetailsComponent />);
    //debug();
    expect(asFragment()).toMatchSnapshot();
  });

  test("Should render Typography", () => {
    const { container } = render(
      <JObDetailsComponent dataTestid="textSample" />
    );
    //debug();
    screen.getAllByTestId("textSample");
    expect(container.querySelectorAll("h6").length).toBe(2);
  });
});
