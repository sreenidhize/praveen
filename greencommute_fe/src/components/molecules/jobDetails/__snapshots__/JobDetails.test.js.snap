// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Details renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    style="width: 390px;"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <hr
        class="MuiDivider-root makeStyles-hrBottom-6"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        class="MuiTypography-root makeStyles-headingClass-1 MuiTypography-body1"
      >
        Description
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h6
        class="MuiTypography-root makeStyles-descClass-4 MuiTypography-h6"
      >
        Open Text is seeking a talented...
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <hr
        class="MuiDivider-root makeStyles-hrTop-5"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        class="MuiTypography-root makeStyles-headingClass-1 MuiTypography-body1"
      >
        What it takes
      </p>
      <ul
        class="makeStyles-ulClass-2"
      >
        <li
          class="makeStyles-liClass-3"
        >
          <h6
            class="MuiTypography-root makeStyles-descClass-4 MuiTypography-h6"
          >
            Excellent written and oral communication and presentation skills
          </h6>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
